services/121-service/src/programs/fsp/fsp.controller.ts

```diff
@@ -76,17 +76,18 @@ export class FspController {
   @ApiResponse({ status: 200, description: 'Vouchers exported' })
   @Post('intersolve/export-voucher')
   public async exportVouchers(
-    @Body() didDto: DidDto,
-    //   @Res() response: Response,
-    // ): Promise<void> {
-  ): Promise<any> {
-    const blob = await this.intersolveService.exportVouchers(didDto.did);
-    // var bufferStream = new stream.PassThrough();
-    // bufferStream.end(Buffer.from(blob, 'binary'));
-    // response.writeHead(200, {
-    //   'Content-Type': 'image/png',
-    // });
-    // bufferStream.pipe(response);
-    return blob;
+    @Body() exportVoucherDto: ExportVoucherDto,
+    @Res() response: Response,
+  ): Promise<void> {
+    const blob = await this.intersolveService.exportVouchers(
+      exportVoucherDto.did,
+      exportVoucherDto.installment,
+    );
+    var bufferStream = new stream.PassThrough();
+    bufferStream.end(Buffer.from(blob, 'binary'));
+    response.writeHead(200, {
+      'Content-Type': 'image/png',
+    });
+    bufferStream.pipe(response);
```

Comment:
I have the feeling that these lines can be simplified using some of the Nest.js/Express build-in methods..

(like `response.contentType('png)`)

But I think we can refactor this in multiple places at once when we get to it...